src/devices/machine/swtpc8212.cpp

```diff
@@ -0,0 +1,475 @@
+
+MC6802P, 2xMC6821P, INS8250N, MCM66750, MC6845P, bank of 8 dips, crystals
+17.0748 (video), 1.8432 (cpu/uart). On the back is a 25-pin RS-232 port, and a
+25-pin printer port.
+
+The 8212 terminal appears similar in design to the CT-82 terminal for which
+there is some documentation and a user guide. In particular it appears closest
+to the 'Version B1'. There was also a 8209 terminal with an 9 inch CRT versus
+the 12 inch CRT in the 8212. This terminal has also been labeled at the
+CT8200.  http://www.swtpc.com/mholley/CT_82/CT82_Index.htm
+
+The 8212 has three CRT controller configurations:
+ 1. 82 x 20 characters, with a height of 14 scan lines.
+ 2. 82 x 24 characters, with a height of 12 scan lines.
+ 3. 92 x 22 characters, with a height of 12 scan lines.
+
+There are two character generators:
+1. The MCM66750 with 7x9 sized characters with some descending by 3 lines
+giving an effective 7x12 sized character set.
+2. An EPROM defining 8x16 sized characters for which only 12 scan lines are
+used, and this is used for the graphics mode.
+
+These appear to map onto the CT-82 screen formats options as follows:
+Format I:   82 by 20, MCM66750 character set. Ctrl-\ crtl-Q
+Format II:  82 by 24, MCM66750 character set. Ctrl-\ crtl-R
+Format III: 82 by 20, EPROM character set.    Ctrl-\ crtl-S
+Format IV:  82 by 24, EPROM character set.    Ctrl-\ crtl-T
+Graphics:   92 by 22, EPROM character set.    Ctrl-] crtl-V
+
+The terminal includes a parallel printer output and Ctrl-] Ctrl-K enables
+printer pass-through and Ctrl-] Ctrl-G disables pass-through.
+
+The terminal appears to include light pen support, and the positions is read
+via Ctrl-] Ctrl-B, TODO.
+
+TODO A generic 'beep' is used. Might want to compare this with the actual 8212
+to better emulate the sound.
+
+TODO A generic keyboard is used. Might need some work to get a better keyboard
+emulation. There also appear to be separate 'cursor' key inputs, with a
+separate strobe, to emulate.
+
+TODO the CB2 input of 'pia0' at address 0x0080 is polled by the firmware, and
+it is the first input checked in the ISR. On a CB2 interrupt the ISR checks
+for a 'break' condition on the UART and if not then it clears the UART OUT1
+output. There are other suggestions in the firmware that the OUT1 and OUT2
+UART lines are being driven. The operation here is unknown?
+
+TODO Confirm the CPU XTAL. The terminal emulation appears slugish compared
+with the documented claims, the hardware flow control is needed even at 9600
+and it does not operate at 38400 as documented. The CT82 manual mentions an
+optional 4MHz crystal and that would run the CPU at 1MHz which is the rated
+speed for the CPU and perhaps the 8212 uses that 4MHz XTAL or at least
+something faster the current 1.8432MHz XTAL?
+
+****************************************************************************/
+
+#include "machine/input_merger.h"
+#include "machine/swtpc8212.h"
+#include "emu.h"
```

Comment:
The PCH and module unit header should come first, otherwise there are going to be problems with different behaviour in different configurations.